controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,294 @@
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info config map.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := odfDeploymentTypeExternal
+	isDROptimized := false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = odfDeploymentTypeInternal
+		if isDROptimized, err = getIsDROptimized(r, storageCluster); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+
+	connectedClients, err := getConnectedClients(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	data := OdfInfoData{
+		Version:           odfVersion,
+		DeploymentType:    odfDeploymentType,
+		IsDROptimized:     isDROptimized,
+		StorageSystemName: storageSystemName,
+		Clients:           connectedClients,
+		StorageCluster: infoStorageCluster{
+			NamespacedName:          client.ObjectKeyFromObject(storageCluster),
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getConnectedClients(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster) ([]connectedClient, error) {
+	storageConsumers := &v1alpha1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumers, client.InNamespace(storageCluster.Namespace))
+	if err != nil {
+		return nil, err
+	}
+	connectedClients := make([]connectedClient, 0, len(storageConsumers.Items))
+
+	for storageConsumerIdx := range storageConsumers.Items {
+		storageConsumer := &storageConsumers.Items[storageConsumerIdx]
+		clusterID := storageConsumer.Status.Client.ClusterID
+		namespacedName := storageConsumer.Status.Client.Name
+		splitNamespaceName := strings.Split(namespacedName, "/")
+		if len(splitNamespaceName) != 2 {
+			return nil, fmt.Errorf("invalid namespaced name %s", namespacedName)
+		}
+
+		namespacedNameObject := types.NamespacedName{
+			Namespace: splitNamespaceName[1],
+			Name:      splitNamespaceName[2],
+		}
```

Comment:
I believe this is going to change after a rebase, now that we moved from a namespace scope clients to cluster scoped clients